Adds AddDropDownButton removes TemplateAddButton

AlexSCorey opened the pull request:
##### SUMMARY
This addresses https://github.com/ansible/awx/issues/5270
Both Inventory List and Template List use the same add button that has a drop down. I decided to make a component that both can use.  This also addresses a typo in a InventoryList test. 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.0
```


##### ADDITIONAL INFORMATION
![Screen Shot 2019-11-08 at 1 20 06 PM](https://user-images.githubusercontent.com/39280967/68500789-8e72ed80-022a-11ea-84f1-577b2408b0e5.png)
![Screen Shot 2019-11-08 at 1 20 30 PM](https://user-images.githubusercontent.com/39280967/68500794-903cb100-022a-11ea-8d1e-bebe3c0df27c.png)

AlexSCorey committed:
Adds AddDropDownButton removes TemplateDropDown Button

Both Inventory List and Template List use the same add button that has a drop down.
I decided to make a component that both can use.
This also addresses a typo in a InventoryList test.

keithjgrant commented on awx/ui_next/src/components/AddDropDownButton/AddDropDownButton.jsx:
```diff
@@ -1,11 +1,10 @@
 import { ToolbarAddButton } from '@components/PaginatedDataList';
 
-function TemplateAddButton({ match, i18n }) {
+function AddDropDownButton({ topUrl, bottomUrl, topLabel, bottomLabel }) {
```
Instead of hard-coding a "top" and "bottom" link, this should take an array of link values, so any number of items can be put in the dropdown. Structure it something like:

```
[{
  url: 'inventory/add',
  label: 'Add Inventory',
}, {
  url: 'smart_inventory/add',
  label: 'Add Smart Inventory',
}]
```

keithjgrant commented on awx/ui_next/src/components/AddDropDownButton/AddDropDownButton.jsx:
```diff
@@ -30,25 +29,21 @@ function TemplateAddButton({ match, i18n }) {
-export { TemplateAddButton as _TemplateAddButton };
-export default withI18n()(withRouter(TemplateAddButton));
+export { AddDropDownButton as _AddDropDownButton };
+export default withI18n()(withRouter(AddDropDownButton));
```
You should be able to remove `withI18n` and `withRouter` here since that has been moved up to the parent component

keithjgrant reviewed (changes requested):
Nice catch cleaning up that chunk of duplicated markup. I just have one suggestion for making this component more general-use

AlexSCorey committed:
Allows AddDropDownButton components to accept array of dropdownItems

AlexSCorey commented:
recheck

keithjgrant reviewed (approved):
Nice work! A couple more thoughts:

* We should probably add propTypes to the TemplateAddButton to help if someone structures the link objects incorrectly
* You could probably get away using `item.url` as the key instead of requiring a key on each object

AlexSCorey commented:
recheck

AlexSCorey committed:
Adds Proptypes to AddDropDownButton Component

Also refactors to use the url that is passed in as the key for the link.
This means that we don't have to pass in a key value.

AlexSCorey commented:
recheck

AlexSCorey commented:
recheck

AlexSCorey committed:
Fixes failing tests

jakemcdermott committed:
Increase async test timeout

When our CI system is overloaded, tests start running slower.

jakemcdermott commented:
recheck

AlexSCorey commented:
recheck

jakemcdermott commented:
recheck

jakemcdermott commented:
recheck
